specification/nginx/Nginx.Management/Certificates.tsp

```diff
@@ -0,0 +1,128 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using Rest;
+using Azure.ResourceManager;
+
+namespace Nginx.NginxPlus;
+
+/** NginxCertificate resource */
+@parentResource(NginxDeployment)
+model Certificate is ProxyResource<NginxCertificateProperties> {
+  ...ResourceNameParameter<Certificate>;
+}
+
+/**
+ * Properties of an Nginx certificate resource.
+ */
+model NginxCertificateProperties {
+  /**
+   * The provisioning state of the certificate.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState;
+
+  /**
+   * The virtual path to the key file.
+   */
+  keyVirtualPath?: string;
+
+  /**
+   * The virtual path to the certificate file.
+   */
+  certificateVirtualPath?: string;
+
+  /**
+   * The Key Vault secret ID.
+   */
+  keyVaultSecretId?: string;
```

Comment:
>string [](http://example.com/codeflow?start=21&length=6)

is it an Azure resource id? If so, model as one!

Reply:
I would basically disagree - if its really already treated as a resource id, in terms of server semantics, then you should just document it, because that's better documentation, and better documentation/specs result in better validations/tooling experiences/error messages etc down the line.

Reply:
Changed it to use the URL scalar